src/compass/s1_geocode_slc.py

```diff
@@ -47,80 +47,81 @@ def run(cfg):
+    for burst in cfg.bursts:
+        date_str = burst.sensing_start.strftime("%Y%m%d")
+        burst_id = burst.burst_id
+        pol = burst.polarization
+        id_pol = f"{burst_id}_{pol}"
+        geo_grid = cfg.geogrids[burst_id]
+
+        # Create top output path
+        burst_output_path = f'{cfg.product_path}/{burst_id}/{date_str}'
+        os.makedirs(burst_output_path, exist_ok=True)
+
+        scratch_path = f'{cfg.scratch_path}/{burst_id}/{date_str}'
+        os.makedirs(scratch_path, exist_ok=True)
+
+        radar_grid = burst.as_isce3_radargrid()
+        native_doppler = burst.doppler.lut2d
+        orbit = burst.orbit
+
+        # Get azimuth polynomial coefficients for this burst
+        az_carrier_poly2d = burst.get_az_carrier_poly()
+
+        # Generate required metadata layers
+        if cfg.rdr2geo_params.enabled:
+            s1_rdr2geo.run(cfg, save_in_scratch=True)
+            if cfg.rdr2geo_params.geocode_metadata_layers:
+               s1_geocode_metadata.run(cfg, fetch_from_scratch=True)
+
+        # Split the range bandwidth of the burst, if required
+        if cfg.split_spectrum_params.enabled:
+            rdr_burst_raster = range_split_spectrum(burst,
+                                                    cfg.split_spectrum_params,
+                                                    scratch_path)
+        else:
+            temp_slc_path = f'{scratch_path}/{id_pol}_temp.vrt'
+            burst.slc_to_vrt_file(temp_slc_path)
+            rdr_burst_raster = isce3.io.Raster(temp_slc_path)
+
+        # Generate output geocoded burst raster
+        geo_burst_raster = isce3.io.Raster(
+            f'{burst_output_path}/{id_pol}.slc',
+            geo_grid.width, geo_grid.length,
+            rdr_burst_raster.num_bands, gdal.GDT_CFloat32,
+            cfg.geocoding_params.output_format)
+
+        # Extract burst boundaries
+        b_bounds = np.s_[burst.first_valid_line:burst.last_valid_line,
+                   burst.first_valid_sample:burst.last_valid_sample]
+
+        # Create sliced radar grid representing valid region of the burst
+        sliced_radar_grid = burst.as_isce3_radargrid()[b_bounds]
+
+        # Geocode
+        isce3.geocode.geocode_slc(geo_burst_raster, rdr_burst_raster,
+                                  dem_raster,
+                                  radar_grid, sliced_radar_grid,
+                                  geo_grid, orbit,
+                                  native_doppler,
+                                  image_grid_doppler, ellipsoid, threshold,
+                                  iters, blocksize, flatten,
+                                  azimuth_carrier=az_carrier_poly2d)
+
+        # Set geo transformation
+        geotransform = [geo_grid.start_x, geo_grid.spacing_x, 0,
+                        geo_grid.start_y, 0, geo_grid.spacing_y]
+        geo_burst_raster.set_geotransform(geotransform)
+        geo_burst_raster.set_epsg(epsg)
+        del geo_burst_raster
+
```

Comment:
```suggestion
        del dem_raster
```

I can't comment on the other earlier part since it wasn't changed in this PR, but here's a fix that worked:
```
(mapping) [staniewi@aurora COMPASS]$ git diff
diff --git a/src/compass/s1_geocode_slc.py b/src/compass/s1_geocode_slc.py
index 83fc26f..b53f74d 100755
--- a/src/compass/s1_geocode_slc.py
+++ b/src/compass/s1_geocode_slc.py
@@ -38,10 +38,6 @@ def run(cfg: GeoRunConfig):
     t_start = time.time()

     # Common initializations
-    dem_raster = isce3.io.Raster(cfg.dem)
-    epsg = dem_raster.get_epsg()
-    proj = isce3.core.make_projection(epsg)
-    ellipsoid = proj.ellipsoid
     image_grid_doppler = isce3.core.LUT2d()
     threshold = cfg.geo2rdr_params.threshold
     iters = cfg.geo2rdr_params.numiter
@@ -50,6 +46,13 @@ def run(cfg: GeoRunConfig):

     # process one burst only
     for burst in cfg.bursts:
+
+        # Reinitialize the dem, since it's modified in geocodeSlc
+        dem_raster = isce3.io.Raster(cfg.dem)
+        epsg = dem_raster.get_epsg()
+        proj = isce3.core.make_projection(epsg)
+        ellipsoid = proj.ellipsoid
+
         date_str = burst.sensing_start.strftime("%Y%m%d")
         burst_id = burst.burst_id
         pol = burst.polarization
@@ -116,6 +119,7 @@ def run(cfg: GeoRunConfig):
         geo_burst_raster.set_geotransform(geotransform)
         geo_burst_raster.set_epsg(epsg)
         del geo_burst_raster
+        del dem_raster # modified in geocodeSlc

         # Save burst metadata
         metadata = GeoCslcMetadata.from_georunconfig(cfg, burst_id)

```